xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/events/AbstractEventStreamEvent.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.eventstream.events;
+
+import org.xwiki.eventstream.Event;
+
+/**
+ * Abstract method for every event that is related to the {@link org.xwiki.eventstream.EventStream}.
+ * When an event that extends an {@link AbstractEventStreamEvent} is triggered, it should also be triggered
+ * with the corresponding event stream event as the source object.
```

Comment:
Well then we should have been way more careful when designing the way the ObservationManager works, because if you take a look at a vast majority of the events defined [here](https://github.com/xwiki/xwiki-platform/tree/master/xwiki-platform-core/xwiki-platform-bridge/src/main/java/org/xwiki/bridge/event), you’ll see that they all come with javadoc such as : 
```
 * The event also send the following parameters:
 * </p>
 * <ul>
 * <li>source: the current {com.xpn.xwiki.doc.XWikiDocument} instance</li>
```

… which, combined with the fact that the [ObservationManager#notify()](https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-observation/xwiki-commons-observation-api/src/main/java/org/xwiki/observation/ObservationManager.java#L90) methods requires at least one object with the event and that [the observation manager isn’t that well documented](https://network.xwiki.com/xwiki/bin/view/DocXCS2En/ObservationModule), is **super confusing**.

---

Anyway … just wanted to say that it’s not obvious. Will be fixed soon.